tensorflow_addons/optimizers/tests/discriminative_layer_training_test.py

```diff
@@ -95,8 +72,78 @@ def test_fit_layer_optimizer(dtype, device, serialize):
     )
 
 
-def test_serialization():
+def test_list_of_layers():
+    model = tf.keras.Sequential(
+        [
+            tf.keras.Input(shape=(4,)),
+            tf.keras.layers.Dense(16),
+            tf.keras.layers.Dense(16),
+            tf.keras.layers.Dense(32),
+            tf.keras.layers.Dense(32),
+        ]
+    )
+
+    optimizers_and_layers = [
+        (tf.keras.optimizers.SGD(learning_rate=0.0), model.layers[0]),
+        (tf.keras.optimizers.Adam(), model.layers[1]),
+        (tf.keras.optimizers.Adam(), model.layers[2:]),
+    ]
+
+    weights_before_train = [
+        [weight.numpy() for weight in layer.weights] for layer in model.layers
+    ]
+
+    multi_optimizer = MultiOptimizer(optimizers_and_layers)
+    model.compile(multi_optimizer, loss="mse")
+
+    x = np.random.rand(128, 4)
+    y = np.random.rand(128, 32)
+    model.fit(x, y, batch_size=32, epochs=10)
+
+    loss = model.evaluate(x, y)
+    assert loss < 0.15
 
+    weights_after_train = [
+        [weight.numpy() for weight in layer.weights] for layer in model.layers
+    ]
+
+    for w_before, w_after in zip(weights_before_train[0], weights_after_train[0]):
+        test_utils.assert_allclose_according_to_type(w_before, w_after)
+
+    for layer_before, layer_after in zip(
+        weights_before_train[1:], weights_after_train[1:]
+    ):
+        for w_before, w_after in zip(layer_before, layer_after):
+            with np.testing.assert_raises(AssertionError):
+                test_utils.assert_allclose_according_to_type(w_before, w_after)
+
+
+def test_model():
+    input = tf.keras.Input(shape=(4,))
+    output = tf.keras.layers.Dense(16)(input)
+    output = tf.keras.layers.Dense(16)(output)
+    output = tf.keras.layers.Dense(32)(output)
+    output = tf.keras.layers.Dense(32)(output)
+    model = tf.keras.Model(input, output)
```

Comment:
Is the case to test a nested model here or do you think is It useless?

Reply:
I recommend having two models to match the common pretraining structure. 

pretrained = get_pretrained_model()

model = tf.keras.Sequential([pretrained, tf.keras.layers.Dense(num_classes)])

Reply:
and just confirm that the pretrained model is not training

Reply:
I meant more something like submodels:
https://github.com/keras-team/keras-io/blob/fc24a909dacc9036d374be5adb168a2daafc8b2c/guides/functional_api.py#L277:L292

Reply:
I believe that is unnecessary based on my understanding on how models become graphs. Both the functional and sequential models are subclasses of the Model class and they should convert to a graph in the same way. I haven't confirmed this, but I'm guessing that sequential is just an wrapper of the functional model. I also haven't ever experienced any differences between functional and sequential models except during model construction. 

However, it wouldn't hurt to put it in.

Reply:
So this loop is ok also in that case?

https://github.com/WindQAQ/addons/blob/d334fabbbee0ae6ef8c35bb124dcec5fee09dd81/tensorflow_addons/optimizers/discriminative_layer_training.py#L150

Reply:
So with current loop we cover also that case?
```weights = [var.name for var in layer.weights]```

Reply:
> So with current loop we cover also that case?
> `weights = [var.name for var in layer.weights]`

Correct. Updated tests.

Reply:
Ops sorry they was the same It was just 1 single case It Is a duplicated comment.